app/src/main/java/me/hufman/androidautoidrive/music/SpotifyMusicMetadata.kt

```diff
@@ -0,0 +1,34 @@
+package me.hufman.androidautoidrive.music
+
+import android.graphics.Bitmap
+import com.spotify.protocol.types.ImageUri
+import me.hufman.androidautoidrive.music.controllers.SpotifyAppController
+
+class SpotifyMusicMetadata(
+		val spotifyController: SpotifyAppController,
+		mediaId: String? = null,
+		queueId: Long? = null,
+		coverArtUri: String? = null,
+		artist: String? = null,
+		album: String? = null,
+		title: String? = null
+		): MusicMetadata(mediaId = mediaId, queueId = queueId, coverArtUri = coverArtUri, artist = artist, album = album, title = title) {
+
+	companion object {
+		fun fromMusicMetadata(spotifyController: SpotifyAppController, musicMetadata: MusicMetadata): SpotifyMusicMetadata {
+			return SpotifyMusicMetadata(spotifyController, mediaId = musicMetadata.mediaId, queueId = musicMetadata.queueId, coverArtUri = musicMetadata.coverArtUri, artist = musicMetadata.artist, album = musicMetadata.album, title = musicMetadata.title)
+		}
+
+		fun createSpotifyMusicMetadataList(spotifyAppController: SpotifyAppController, musicMetadataList: List<MusicMetadata>): List<SpotifyMusicMetadata> {
+			val spotifyMusicMetadataList = ArrayList<SpotifyMusicMetadata>()
+			musicMetadataList.forEach {
```

Comment:
Use a map instead: `return musicMetadataList.map { fromMusicMetadata(spotifyAppController, it) }`

Reply:
That is so much cleaner, I haven't used the map method much in Kotlin before so I need to start apparently!

Reply:
Kotlin is a great mix of functional constructs from other languages (like Ruby) but integrates so nicely into the JVM!